packages/vite/src/node/plugins/resolve.ts

```diff
@@ -553,8 +554,12 @@ function tryResolveFile(
     }
   }
 
-  const tryTsExtension = options.isFromTsImporter && isPossibleTsOutput(file)
-  if (tryTsExtension) {
+  // try resolve .js import to typescript file
+  if (
+    !skipTsExtension &&
```

Comment:
I'm not sure this is enough.

The issue is that the resolution goes something like that:
  - .js
    - .ts
    - .tsx
  - .jsx
    - . tsx
  - .ts
  - .tsx
  
I think the list of extensions to check should be determine in the previous scope, which would avoid trying `.ts`
extension twice when the file was extension-less at the beginning

Reply:
Ah I didn't intend to fix that initially, but it seems like a simple fix. I think b568012 should do it so it avoids the nested resolve.